### PR TITLE
Fix/more rs fixes

### PR DESCRIPTION
**Pull request by Raycoms**
Fix a crash with the request system + command permission + clipboard recipe

Review please

### CODE REVIEW

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 43 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L416](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L416): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L416](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L416): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L416](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L416): Refactor this method to reduce its Cognitive Complexity from 29 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render#](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/a20c5b51c46a837444dcb6a91b94bda08387dbc3/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 48 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 43 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Refactor this method to reduce its Cognitive Complexity from 29 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render#](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/96d8373b0e5656b871667f8ec8a0cb226aeb11d0/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
looks good to me :shipit: (I'll delete my PR, didn't realise you already made a PR with a fix)

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1788/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import com.minecolonies.api.util.Log;
 import com.minecolonies.api.util.Vec2i;
+import com.minecolonies.api.util.constant.Constants;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java:5]: MINOR Remove this unused import 'com.minecolonies.api.util.constant.Constants'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 73 issues
* ![MAJOR][MAJOR] 15 major
* ![MINOR][MINOR] 56 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L349): Remove this unused private "getNicePositionString" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): class "AbstractEntityAIBasic" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L240](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L240): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L474](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L474): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L714](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L714): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/a8cd4b1ac5bc4ef0ff16b75d4539547d154f8ac0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 89 issues
* ![MAJOR][MAJOR] 27 major
* ![MINOR][MINOR] 57 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L349): Remove this unused private "getNicePositionString" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): class "AbstractEntityAIBasic" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L240](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L240): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L474](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L474): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L714](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L714): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/613885f0982718e7dc605491bb19b9a613132bab/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 89 issues
* ![MAJOR][MAJOR] 27 major
* ![MINOR][MINOR] 57 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L349): Remove this unused private "getNicePositionString" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): class "AbstractEntityAIBasic" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L240](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L240): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L474](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L474): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L714](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L714): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/04d613fd46cb3d328a0debe075ea517280166d5e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/client/gui/WindowRequestDetail.java [hunk 2]
```diff
@@ -181,9 +183,18 @@ public void onOpened()
...
-        findPaneOfTypeByID(RESOLVER, Label.class).setLabelText("Resolver: " +
-                                                                 ColonyManager.getColony(colonyId).getRequestManager().getResolverForRequest(
-                                                                   request.getToken()).getDisplayName(request.getToken()).getFormattedText());
+        try
+        {
+            findPaneOfTypeByID(RESOLVER, Label.class).setLabelText("Resolver: " +
+                    ColonyManager.getColony(colonyId).getRequestManager().getResolverForRequest(
+                            request.getToken()).getDisplayName(request.getToken()).getFormattedText());
+        }
+        catch(@SuppressWarnings(EXCEPTION_HANDLERS_SHOULD_PRESERVE_THE_ORIGINAL_EXCEPTIONS) final IllegalArgumentException e)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowRequestDetail.java:192]: MAJOR Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 90 issues
* ![MAJOR][MAJOR] 28 major
* ![MINOR][MINOR] 57 minor
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WindowCitizen.java#L349](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L349): Remove this unused private "getNicePositionString" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L55](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L55): class "AbstractEntityAIBasic" has 53 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L240](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L240): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L474](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L474): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L714](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L714): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L78](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L78): Remove this unused "wareHouse" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L130](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L130): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [EntityAIWorkDeliveryman.java#L413](https://github.com/Minecolonies/minecolonies/blob/8f8bce5b03fb7eb9ab6557e477533f4014e2ae72/src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java#L413): The Cyclomatic Complexity of this method "deliver" is 19 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**